core-api/src/main/java/com/optimizely/ab/config/audience/match/SemanticVersionAttributeMatch.java

```diff
@@ -0,0 +1,67 @@
+/**
+ *
+ *    Copyright 2020, Optimizely and contributors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.optimizely.ab.config.audience.match;
+
+abstract class SemanticVersionAttributeMatch implements Match {
+    public int compareVersion(String actualVersion, String targetVersion) {
+        if (actualVersion.equals(targetVersion)) {
+            // Any version.
+            return 0;
+        }
+
+        // Expect a version string of the form x.y.z-(string)
+        String[] actualVersionParts = actualVersion.split("[-\\.]");
+        String[] targetVersionParts = targetVersion.split("[-\\.]");
+
+        // Check only till the precision point of actualVersionParts
+        for (int i = 0; i < actualVersionParts.length; i++) {
```

Comment:
we should check using targetedVersionParts instead of actual. Use proper name targetIndex, instead of `i`